src/utils/consoleColor.lua

```diff
@@ -0,0 +1,45 @@
+local ANSI_COLOUR_TEMPLATE = "\x1b[%s;%s;%sm%s\x1b[0;0;0m"
+local DEFAULT_BG_CODE = 49
+
+--- Creates a function which takes in text and colours it according to the arguments.
+---@param modifier number 0 means normal, 1 means bold and 2 means dimmed
+---@param colourCode number ANSI escape colour code
+---@param bgColourCode number|nil Optional background colour code
+---@return fun(text:string|nil):string textFn
+local function createColouredTextFn(modifier, colourCode, bgColourCode)
+	return function(text)
+		text = text or ""
```

Comment:
I'm not sure this should allow using `nil` like this, do you have an example use case?